plugins/woocommerce-admin/client/products/sections/utils.ts

```diff
@@ -1,110 +1,79 @@
 /**
  * External dependencies
  */
-import classnames from 'classnames';
 import { recordEvent } from '@woocommerce/tracks';
 
 /**
  * Internal dependencies
  */
 import { NUMBERS_AND_ALLOWED_CHARS } from '../constants';
 
-type gettersProps = {
-	context?: {
-		formatAmount: ( number: number | string ) => string;
-		getCurrencyConfig: () => {
-			code: string;
-			symbol: string;
-			symbolPosition: string;
-			decimalSeparator: string;
-			priceFormat: string;
-			thousandSeparator: string;
-			precision: number;
-		};
-	};
-	value: string;
-	name?: string;
-	checked: boolean;
-	selected?: boolean;
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	onChange: ( value: any ) => void;
-	onBlur: () => void;
-	className: string | undefined;
-	help: string | null | undefined;
+type CurrencyConfig = {
+	code: string;
+	symbol: string;
+	symbolPosition: string;
+	decimalSeparator: string;
+	priceFormat: string;
+	thousandSeparator: string;
+	precision: number;
 };
 
-export const getCheckboxProps = ( {
-	checked = false,
-	className,
-	name,
-	onBlur,
-	onChange,
-}: gettersProps ) => {
+/**
+ * Get additional props to be passed to all checkbox inputs.
+ *
+ * @param {string} name Name of the checkbox
+ * @return {object} Props.
+ */
+export const getCheckboxProps = ( name: string ) => {
 	return {
-		checked,
-		className: classnames( 'woocommerce-product__checkbox', className ),
 		onChange: ( isChecked: boolean ) => {
 			recordEvent( `product_checkbox_${ name }`, {
 				checked: isChecked,
 			} );
-			return onChange( isChecked );
 		},
-		onBlur,
 	};
 };
 
-export const getTextControlProps = ( {
-	className,
-	onBlur,
-	onChange,
-	value = '',
-	help,
-}: gettersProps ) => {
+/**
+ * Get input props for currency related values and symbol positions.
+ *
+ * @param {object}  context - Currency context
+ * @return {object} Props.
+ */
+export const getCurrencyInputProps = ( currencyConfig: CurrencyConfig ) => {
+	const { symbol, symbolPosition } = currencyConfig;
+	const currencyPosition = symbolPosition.includes( 'left' )
+		? 'prefix'
+		: 'suffix';
+
 	return {
```

Comment:
My goal was to unwrap some of these so that we can:
1. More easily follow where props are being passed
2. Reduce the knowledge of form handlers in this function
3. Reduce overall logic in this function